plugins/Jobs.cpp

```diff
@@ -295,12 +295,17 @@ bool BedrockJobsCommand::peek(SQLite& db) {
                 }
 
                 // If there's no job or the existing job doesn't match the data we've been passed, escalate to leader.
-                if (!result.empty() && ((job["data"].empty() && result[0][1] == "{}") || (!job["data"].empty() && result[0][1] == job["data"]))) {
-                    // Return early, no need to pass to leader, there are no more jobs to create.
-                    SINFO("Job already existed and unique flag was passed, reusing existing job " << result[0][0] << ", mocked? "
-                      << (mockRequest ? "true" : "false"));
-                    jsonContent["jobID"] = result[0][0];
-                    return true;
+                if (!result.empty()) {
+                    if (result[0][2] != "0" && result[0][2] != job["parentJobID"]) {
+                        STHROW("404 Trying to create a child that already exists, but it is tied to a different parent");
```

Comment:
If you do this, it's going to fail for all jobs, not only the one that is tied to a different parent, are we ok with it?

Reply:
heh yes, I said this in our thread:
> I think my code is correct anyway... just reject everything if one of the things you are doing is completely invalid

I am totally fine rejecting everything if what you are trying to do includes something that makes no sense. In fact it's probably better we do this than the alternative, because if we create just part of the jobs, then how do we even return the fact that one was completely invalid?